pkg/istio-agent/agent.go

```diff
@@ -205,6 +210,35 @@ func NewAgent(proxyConfig *mesh.ProxyConfig, agentOpts *AgentOptions, sopts *sec
 		cfg:         agentOpts,
 		secOpts:     sopts,
 		envoyOpts:   eopts,
+		watcher:     filewatcher.NewWatcher(),
+		tlsOpts:     &citadel.TLSOptions{},
+	}
+}
+
+func (a *Agent) watcherHandler(ctx context.Context) {
+	for {
+		select {
+		case gotEvent := <-a.watcher.Events(a.tlsOpts.RootCert):
+			log.Infof("Receive file %s event %v", a.tlsOpts.RootCert, gotEvent)
+			if a.caClient != nil {
+				err := a.caClient.InitClient()
+				if err != nil {
+					log.Warnf("Failed to init CA client")
+					continue
+				}
+
+				a.secretCache.CallUpdateCallback(security.RootCertReqResourceName)
```

Comment:
Strictly speaking, `a.tlsOpts.RootCert` and `RootCertReqResourceName` are not the same thing. The former is the root cert of *istiod*, and the latter the root cert *of the mesh* - they may be different.

And we already have a mechanism to update the latter - https://github.com/istio/istio/blob/3083ead98573a5305e98566650d3637fb6b4ae39/pilot/pkg/xds/pcds.go#L67. So we may need to consider if we want 2 ways to do this. I think its reasonable but curious of @shankgan opinion

Reply:
I'm confused on this point. After restart istiod pod, the connection with sidecar is terminated.  can xds work on that time?

Reply:
You can safely rely on pcds then